pose_format/tensorflow/masked/tensor.py

```diff
@@ -121,3 +125,21 @@ def rename(self, *names):
         tensor = self.tensor.rename(*names)
         mask = self.mask.rename(*names)
         return MaskedTensor(tensor=tensor, mask=mask)
+
+    def mean(self, axis=None):
+        mt_sum = tf.math.reduce_sum(self.zero_filled(), axis=axis)
+        mt_count = tf.math.reduce_sum(tf.cast(self.mask, mt_sum.dtype), axis=axis)
+        tensor = tf.math.divide(mt_sum, mt_count)
+        mask = tf.cast(mt_count, tf.bool)
+        mt = MaskedTensor(tensor=tensor, mask=mask)
```

Comment:
Here I am not sure why a sum value should be masked if the result of summation is 0. Or am I misunderstanding what is happening?

Reply:
we sum the tensor, and *count* how many values were in every aggregated axis.
for example, `[[1, 2], [3, 4]]` masked with `[[1, 1], [0, 1]]` over axis 1 would be zerofilled to `[[1,2],[0,4]]`, then summed to `[1, 6]` and counted `[1, 2]`. we then divide = `[1, 3]`.

Now, we could return a `tf.Tensor`, in that case, but what if the count was `[0,2]`, then we would have `[nan, 3]` which requires masking, at the very least in order to fix the `nan` values.

this is also the definition of `ma.mean`, it would return a masked array of that sort.

Reply:
Got it!